server/app/auth/oidc/CustomOidcLogoutRequest.java

```diff
@@ -0,0 +1,87 @@
+package auth.oidc;
+
+import com.google.common.base.Strings;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import java.net.URI;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CustomOidcLogoutRequest extends LogoutRequest {
+  /** The optional post-logout redirection param and URI. */
+  private final String postLogoutRedirectParam;
+
+  private final URI postLogoutRedirectURI;
+
+  private final Map<String, String> extraParams;
```

Comment:
Can we use ImmutableMap here to provide a further guarantee that the key/values in the map itself are unmodifiable?